pkg/blockstorage/awsefs/awsefs.go

```diff
@@ -61,13 +62,23 @@ func NewEFSProvider(config map[string]string) (blockstorage.Provider, error) {
 		return nil, errors.New("Account ID is empty")
 	}
 	accountID := *user.Account
-	efsCli := awsefs.New(s, aws.NewConfig().WithRegion(region))
-	backupCli := backup.New(s, aws.NewConfig().WithRegion(region))
+	creds := awsConfig.Credentials
+	if roleName != "" {
+		role := roleARN(*user.Account, roleName)
+		creds = stscreds.NewCredentials(s, role)
+	}
+	efsCli := awsefs.New(s, aws.NewConfig().WithRegion(region).WithCredentials(creds))
+	backupCli := backup.New(s, aws.NewConfig().WithRegion(region).WithCredentials(creds))
 	return &efs{
 		EFS:       efsCli,
 		Backup:    backupCli,
 		region:    region,
-		accountID: accountID}, nil
+		accountID: accountID,
+	}, nil
+}
+
+func roleARN(accountID, roleName string) string {
+	return fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, roleName)
```

Comment:
I think we should accept "roleArn" as an input instead of roleName. It is valid to specify a role that is in a different account